src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -74,6 +76,25 @@ public static class HostUtilities
 
         private static readonly List<Hashtable> s_suggestions = InitializeSuggestions();
 
+        private static bool HostSupportUnicode()
+        {
+            // Reference: https://github.com/zkat/supports-unicode/blob/main/src/lib.rs
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return Environment.GetEnvironmentVariable("WT_SESSION") is not null ||
+                    Environment.GetEnvironmentVariable("TERM_PROGRAM") is "vscode" ||
+                    Environment.GetEnvironmentVariable("ConEmuTask") is "{cmd:Cmder}" ||
+                    Environment.GetEnvironmentVariable("TERM") is "xterm-256color" or "alacritty";
```

Comment:
Should it be caseinsensetive?

Reply:
I prefer to stick with the reference. Those env variables should be declared by the terminals, in the exact names.